tutorials/basic-topics/sorting/insertion-sort.md

```diff
@@ -11,4 +11,222 @@ keywords:
+* First of all, we will pick the first element which is $3$ in our case.
+* Now, we will compare with all the elements left to it, in this case there is nothing left to $3$ so, we will do nothing.
+* Now, pick the next element which is $2$ and start comparing with all the elements left to it.
+* First element left to $2$ is $3$ , which is greater than selected element $2$. So, we will move $3$ to right. If the element left to selected element is less than selected element then, our selected element have reached it's correct position so, we will place selected element there only.
+* Now, there is nothing no more element further left to our selected element $2$. So, we will place $2$ at the beginning of array.
+* We will proceed with the same way for all the elements. Then, at last we will get our array as sorted.
+
+### Problem Link
+
+https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/
+
+### Problem Statement
+
+Given the array of integers nums, you will choose two different indices i and j of that array. Return the maximum value of $(nums[i] - 1) * (nums[j] - 1)$.
+
+**Example 1:**
+
+```
+Input: nums = [3,4,5,2]
+Output: 12 
+Explanation: If you choose the indices i=1 and j=2 (indexed from 0), you will get the maximum value, that is, (nums[1]-1)*(nums[2]-1) = (4-1)*(5-1) = 3*4 = 12. 
+```
+
+**Example 2:**
+
+```
+Input: nums = [1,5,4,5]
+Output: 16
+Explanation: Choosing the indices i=1 and j=3 (indexed from 0), you will get the maximum value of (5-1)*(5-1) = 16.
+```
+
+**Example 3:**
+
+```
+Input: nums = [3,7]
+Output: 12
+```
+
+**Constraints:**
+
+* `2 <= nums.length <= 500`
+* `1 <= nums[i] <= 10^3`
+
+## Approach: Sorting using Insertion Sort
+
+The problem wants us to find the maximum value of $(nums[i] - 1) * (nums[j] - 1)$, where $i$ and $j$ are two different indices of the given array.
+
+By looking at the expression, we can observe that its just a product of two numbers of the given array.
+
+We have to maximize the product. Now, in order to maximize the product of two numbers we have to choose the two largest number possible. So, in order to find the maximum value of $(nums[i] - 1) * (nums[j] - 1)$ we just have to take the largest two number in the given array.
+
+Now, we know how can we maximize the value of given expression. But we have to also figure out the how can we get the two largest number present in the given array. Here comes the sorting method, if we sort our array then, the largest number would be present at the last index and second largest number would be present at the second last index of the sorted array. For sorting, we are going to use Insertion Sort Algorithm.
+
+We have figured out the solution of the problem:
+* Sort the given array using insertion sort (Refer to the algorithm section).
+* Take out the last two elements because those are the largest two elements in our array.
+* Put the values in the expression and return it.
+
+## Implementation
```

Comment:
We don't need to say Here is the implementation. You may remove this part as well as line 97.
@wingkwong Any inputs from your side ?

Reply:
Agree.

Reply:
Removed L95-L97.